microprot/scripts/processing.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+__author__ = "Tomasz Kosciolek"
+__version__ = "1.01b"
+__last_update__ = "12/19/2016"
+
+
+import click
+import sys
+from skbio import io, Sequence
+
+
+def extract_sequences(infile, seqidx=0):
+    """ Extract sequence(s) from a multi-sequence FASTA file
+
+    Parameters
+    ----------
+    infile : str
+        file path to input multi-sequence FASTA file
+    seqidx : int (optional)
+        n-th sequence of the input file to extract
+        (default: 0 for all sequences)
+
+    Returns
+    -------
+    list of skbio Sequence
+    """
+    seqs = []
+    iseq = 0  # current sequence index
+    for seq in io.read(infile, format='fasta'):
+        iseq += 1
+        if 0 < seqidx != iseq:
+            continue
```

Comment:
You are right! `break` after the first match will save some time. But meanwhile I have two questions:
1) I found that I couldn't directly access an element of the `io` object. For example, I cannot do:
```
seqs = io.read(infile, format='fasta')
return seqs[seqidx]
```
Which is unfortunate. Is there a better way? @mortonjt @RNAer 

2) I guess that in the near future, this function will not be limited to one sequence index, but multiple
(e.g., `extract_sequences('input.fasta', seqidx=1,2,3,6,7,11)`
or `extract_sequences('input.fasta', seqidx='ids.txt')`.
Therefore I hesitate if we should `break` after the match. What do you think? @tkosciol